curriculum/challenges/english/25-front-end-development/lab-email-masker/66b205e6eacba4c4e54ea434.md

```diff
@@ -0,0 +1,102 @@
+---
+id: 66b205e6eacba4c4e54ea434
+title: Build an Email Masker
+challengeType: 14
+dashedName: build-an-email-masker
+---
+
+# --description--
+
+In this lab, you will mask the username part of an email address with asterisks. Masking is a term used to hide or replace sensitive information with asterisks or other characters.
+
+For example, if the email address was `myEmail@email.com`, then the masked email address will be `m*******l@email.com`.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. Create a function named `maskEmail` that takes `email` as an argument.
+2. Inside the function, you should create a variable called `maskedName` and assign it the result of masking the `email` along with concatenating the domain name.  Remember that you can use methods like `slice`, `repeat`, `indexOf` `slice` or even `replace` to help you. Return `maskedName` from the function.
```

Comment:
I agree with Tom here about the maskedName.
We were using that earlier because this lab originally didn't have functions.
But now that is does, we don't need it any more

Reply:
Thank you for this suggestion, Tom and Jessica. 
We won't need that user story now. I have updated the PR.